src/serial.rs

```diff
@@ -117,6 +117,8 @@ pub trait SerialEvents {
     fn out_byte(&self);
     /// An error occurred while writing a byte to serial output resulting in a lost byte.
     fn tx_lost_byte(&self);
+    /// The input buffer is empty.
```

Comment:
I'd like to see more in the doc comment here. 

I think it's useful to let people know how this can be used. For example, we can write something in the lines of: "This event can be used by the consumer to re-enable events coming from the serial input."